File: summer/WebContent/js/system/windows/controls/GroupBox.js
/**
 * GroupBox
 */

define(["dojo/_base/declare", "system/Type", "controls/HeaderedContentControl", "input/TraversalRequest",
        "input/FocusNavigationDirection"], 
		function(declare, Type, HeaderedContentControl, TraversalRequest,
				FocusNavigationDirection){
	var GroupBox = declare("GroupBox", HeaderedContentControl,{
		constructor:function(){
		},
		
        /// <summary> 
        /// The Access key for this control was invoked. 
        /// </summary>
//        protected override void 
        OnAccessKey:function(/*AccessKeyEventArgs*/ e) 
        {
            this.MoveFocus(new TraversalRequest(FocusNavigationDirection.First));
        }
	});
	
//    private static void 
    function OnAccessKeyPressed(/*object*/ sender, /*AccessKeyPressedEventArgs*/ e)
    { 
        if (!e.Handled && e.Scope == null && e.Target == null) 
        {
            e.Target = sender instanceof GroupBox ? sender : null; 
        }
    }
    
//    static GroupBox() 
    function Initialize()
    {
    	UIElement.FocusableProperty.OverrideMetadata(GroupBox.Type,
    			/*new FrameworkPropertyMetadata(false)*/FrameworkPropertyMetadata.BuildWithDV(false)); 
    	Control.IsTabStopProperty.OverrideMetadata(GroupBox.Type, 
    			/*new FrameworkPropertyMetadata(false)*/FrameworkPropertyMetadata.BuildWithDV(false)); 
    	FrameworkElement.DefaultStyleKeyProperty.OverrideMetadata(GroupBox.Type, 
    			/*new FrameworkPropertyMetadata(GroupBox.Type)*/FrameworkPropertyMetadata.BuildWithDV(GroupBox.Type));
//        EventManager.RegisterClassHandler(GroupBox.Type, AccessKeyManager.AccessKeyPressedEvent, 
//        		new AccessKeyPressedEventHandler(null, OnAccessKeyPressed)); 
    };
	
	GroupBox.Type = new Type("GroupBox", GroupBox, [HeaderedContentControl.Type]);
	Initialize();
	
	return GroupBox;
});

File: org.summer.view.widget/src/org/summer/view/widget/controls/GroupBox.java
/**
 */
package org.summer.view.widget.controls;

import org.summer.view.widget.EventManager;
import org.summer.view.widget.FrameworkPropertyMetadata;
import org.summer.view.widget.input.AccessKeyEventArgs;
import org.summer.view.widget.input.AccessKeyManager;
import org.summer.view.widget.input.AccessKeyPressedEventArgs;
import org.summer.view.widget.input.FocusNavigationDirection;
import org.summer.view.widget.input.TraversalRequest;
import org.summer.view.window.automation.peer.AutomationPeer;


/// <summary>
/// GroupBox Control class 
/// </summary> 
//[Localizability(LocalizationCategory.None, Readability = Readability.Unreadable)] // cannot be read & localized as string
public class GroupBox extends HeaderedContentControl 
{
//    #region Constructors

    static/* GroupBox()*/ 
    {
        FocusableProperty.OverrideMetadata(typeof(GroupBox), new FrameworkPropertyMetadata(false)); 
        IsTabStopProperty.OverrideMetadata(typeof(GroupBox), new FrameworkPropertyMetadata(false)); 
        DefaultStyleKeyProperty.OverrideMetadata(typeof(GroupBox), new FrameworkPropertyMetadata(typeof(GroupBox)));
        EventManager.RegisterClassHandler(typeof(GroupBox), AccessKeyManager.AccessKeyPressedEvent, new AccessKeyPressedEventHandler(OnAccessKeyPressed)); 
    }

//    #endregion

//    #region Override methods

    /// <summary> 
    /// Creates AutomationPeer (<see cref="UIElement.OnCreateAutomationPeer"/>)
    /// </summary> 
    protected /*override*/ AutomationPeer OnCreateAutomationPeer()
    {
        return new System.Windows.Automation.Peers.GroupBoxAutomationPeer(this);
    } 

    /// <summary> 
    /// The Access key for this control was invoked. 
    /// </summary>
    protected /*override*/ void OnAccessKey(AccessKeyEventArgs e) 
    {
        MoveFocus(new TraversalRequest(FocusNavigationDirection.First));
    }

    private static void OnAccessKeyPressed(Object sender, AccessKeyPressedEventArgs e)
    { 
        if (!e.Handled && e.Scope == null && e.Target == null) 
        {
            e.Target = sender as GroupBox; 
        }
    }

//    #endregion 
}